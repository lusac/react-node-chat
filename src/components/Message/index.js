import React from 'react'
import PropTypes from 'prop-types'

export default class Message extends React.Component {
  render() {
    return (
      <div className="message">
        <strong className="message__username">
          {this.props.message.username}
        </strong>

        <span className="message__text">
          {this.props.message.text}
        </span>
      </div>
    )
  }
}

Message.propTypes = {
  message: PropTypes.object
}