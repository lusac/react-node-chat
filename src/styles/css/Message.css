.message {
  display: flex;
  padding: 18px 0;
  flex-direction: column;
  border-bottom: 2px dotted #CCC;
}
.message__username {
  color: #666;
  font-size: 14px;
}
.message__text {
  color: #333;
  font-size: 15px;
}

/*# sourceMappingURL=Message.css.map */
