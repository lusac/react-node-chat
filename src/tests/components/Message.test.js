/* global it describe expect */
import React from 'react'
import { shallow, configure } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Message from '../../components/Message'

configure({ adapter: new Adapter() })

describe('Message component', () => {
  describe('snapshot', () => {
    it('channel not selected', () => {
      const
        message = {username: 'xunda', text: 'hello!', date: new Date()},
        renderedValue =  renderer.create(<Message message={message} />).toJSON()
      expect(renderedValue).toMatchSnapshot()
    })
  })

  it('formatDate method', () => {
    let
      message = {username: 'xunda', text: 'hello!', date: new Date()},
      wrapper = shallow(<Message message={message}/>),
      date = wrapper.instance().formatDate(new Date('Mon Aug 27 2018 11:22:33'))
    expect(date).toBe('11:22 AM')
  })
})