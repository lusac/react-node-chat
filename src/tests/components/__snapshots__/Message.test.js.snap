// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component snapshot channel not selected 1`] = `
<div
  className="message"
>
  <div
    className="message__left"
  >
    <span
      className="message__photo"
    >
      X
    </span>
  </div>
  <div
    className="message__right"
  >
    <strong
      className="message__info"
    >
      <span>
        xunda
      </span>
      <span
        className="message__date"
      >
        8/27/2018, 11:22 AM
      </span>
    </strong>
    <span
      className="message__text"
    >
      hello!
    </span>
  </div>
</div>
`;
